### ISSUE TITLE
folder text view 3000+ words renders manuskript unresponsive

### ISSUE THREAD

**Issue by bartart3d** [olivierkes/manuskript#595]
Hi all,
I wanted to read a chapter, bundled in a folder containing all the scenes in separate files of about 500 words. So I clicked on the icon to show the text instead of the tiles with summaries. 
Manuskript instantly grinded to a halt. The text started to show up only after a few seconds, and scrolling through it became impossible. Clicking back to the tiles did work, but again, it was so slow and unresponsive I had to wait several seconds to let it happen.
Is it a memory issue, or a rendering issue?

grtz,
Bart

**Comment by worstje** [comment]
That doesn't sound familiar. I was going to suggest it might be a hard drive issue, but then I remembered Manuskript loads files into memory when you first load the project, so that can't be it.

So a memory issue is most likely. Still, 500 words or 3,000 words are nothing, and you shouldn't be seeing any slow downs unless we are talking millions of words spread out over hundreds of files.

Can you provide a step-by-step guide that will produce the issue? If we can replicate it we can give you more feedback.

**Comment by bartart3d** [comment]
Keeping the folder in text mode open for a while (30 seconds to a minute) makes manuskript responsive again. 

Reproducing seems not possible on newly created file here either, but I do get crashes while quickly copy pasting lorem ipsum generated text into the new text files.
To try to reproduce: 
- create new document with chapters (folders)
- fill the text files within a chapter with text
- click on the chapter (folder)
- select text view instead of the two other options.

The problem appeared on a side-project, so I tested it on my main project to check if there wasn't some kind of file corruption because of a faulty drive (it's on a different drive). To my surprise manuskript hung up too when clicking on text mode for a folder. Even worse, it crashed, and now... each time I open the file, manuskript crashes when I barely move the mouse... weird stuff... 

Maybe something python related? I upgraded from 18.10 to 19.04, maybe a dependency is not met to make it run properly... ?

**Comment by bartart3d** [comment]
I hardly can believe it's a memory issue, I edit video and audio, eating up GB's of memory without a problem. I'll try a memtest in the morning.

**Comment by bartart3d** [comment]
Ok, new progress.. with a question.. is it somehow possible that manuskript is influenced from one file to the other? I mean, I just solved a thing in one file, by changing something in the other file, or so it appears.
The folder of the sideproject was set to text view, and it did load, but slowly. That's what this issue is about. I saved it. Then I closed manuskript, opened up the main project, which crashed when moving the mouse. Running from terminal it only shows segmentation fault. Now I set the view in the side project back to the tiles, saved it and, to give it another shot, I opened the main project and now it does not crash anymore when I move the mouse. This is the only thing I have changed. It seems impossible to me that both issues are related, but with this happening...

**Comment by Eusorph** [comment]
I do get this problem both on single text files longer than 3000 words and on folders that contain broken up text that sums up over that amount. To avoid the problem, I have in fact broken up my chapters in scenes shorter than 3000 words and never click on folders, which means I can never user folder text view. It happens always, on newly created files (while I was  putting in my 135000 novel) or on the old files I am changing by making revisions.

**Comment by getinstancemz** [comment]
I have the same issue. As long as I click only only on files -- I write one scene per file, and my scenes are relatively short  -- I'm fine. But occasionally I'll unselect, or hit a folder containing too many scenes. Then it's time to go and make coffee.

**Comment by Zadeis** [comment]
I want to add some potential data that I hope will help with the subject issue. My wife started using manuskript recently and seems to be hitting the described problem. I ran cProfile for the given amount of time it says in the attached file. I hope it may shine some light on the problem. If you need me to run it again please say so.

[cProfile_out.txt](https://github.com/user-attachments/files/17544656/cProfile_out.txt)

I am running this on linux via the bin folder of the source code for the current release of 0.16.1 (not any of the current branches, but from the unzipped source code) as described in the wiki. The command used to run it is as follows:

python -m cProfile manuskript > cProfile_out.txt

If the authors feel that the content of the file from cProfile is relevant to the subject problem just say so and I will open a separate issue.
